package/MDAnalysis/converters/OpenMMParser.py

```diff
@@ -123,24 +133,55 @@ def _mda_topology_from_omm_topology(self, omm_topology):
         bond_orders = [b.order for b in omm_topology.bonds()]
         bond_types = [b.type for b in omm_topology.bonds()]
 
-        n_atoms = len(atomids)
-        n_residues = len(resids)
-        n_segments = len(segids)
-
         attrs = [
             Atomids(np.array(atomids, dtype=np.int32)),
             Atomnames(np.array(atomnames, dtype=object)),
-            Atomtypes(np.array(atomtypes, dtype=object)),
             Bonds(bonds, types=bond_types, order=bond_orders, guessed=False),
             ChainIDs(np.array(chainids, dtype=object)),
-            Elements(np.array(elements, dtype=object)),
-            Masses(np.array(masses, dtype=np.float32)),
             Resids(resids),
             Resnums(resnums),
             Resnames(resnames),
             Segids(segids),
         ]
 
+        validated_elements = []
+        masses = []
+        atomtypes = []
+        for a in omm_topology.atoms():
+            elem = a.element
+            if elem is not None and elem.symbol.capitalize() in SYMB2Z:
+                validated_elements.append(elem.symbol.capitalize())
+                masses.append(elem.mass._value)
+                atomtypes.append(elem.symbol.capitalize())
+            else:
+                wmsg = (f"Unknown element {elem} found for some atoms. "
+                        f"These have been given an empty element record "
+                        f"with their atomtype set to 'X' "
+                        f"and their mass set to 0.0. "
+                        f"If needed they can be guessed using "
+                        f"MDAnalysis.topology.guessers.")
+                warnings.warn(wmsg)
+                validated_elements.append('')
+                masses.append(0.0)
+                atomtypes.append('X')
```

Comment:
I know we said to use an "X" in AtomTypes for empty elements, but could you please actually use the elem.symbol in Atomtypes? There could be weird elements in there and it would be best to keep that information. A relevant test would probably require manually constructing a small OpenMM topology with manually defined elements like below:

```
>>> from openmm.app.element import Element
>>> Element(0, "*", "*", 0)
<Element *>

```

Reply:
Sorry I've been offline for many days, was busy with exams. 
re - using elem.symbol, for this case what if the elem itself is None? I've changed my code to accept weird types that are not in SYMB2Z but when the elem is None, should we stick to using 'X', should we guess it or do something else?

Reply:
I think the logic flow should probably go like this to be consistent with other parsers and not lose information -- (@IAlibay you're probably more in touch with this than I am, any comments or emoji reacts?):

- if element is present:
  - if element is valid, use that symbol for both AtomTypes and Elements
  - if element is invalid, use the symbol for AtomType but have an empty '' for Element
- if element is absent:
  - use 'X' for AtomType and empty '' for Element

Reply:
Yeah that sounds reasonable